src/app/lab/components/units-dropdown.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, Input } from '@angular/core';
+
+import { Chart } from '../../models/chart.model';
+
+/*  Units Dropdown Component
+
+    -- Requires project input
+    Expected use:
+        <ccl-units-dropdown
+            [chart]="your_chart">
+*/
+
+@Component({
+  selector: 'ccl-units-dropdown',
+  template: `<div dropdown class="dropdown dropdown-units">
+              <button dropdownToggle class="button dropdown-toggle" type="button"
+                id="unitsDropdown" data-toggle="dropdown" aria-haspopup="true"
+                aria-expanded="true">
+                <i class="icon-flash"></i>
+                {{ chart.unit }}
+                <i class="icon-angle-down caret"></i>
+              </button>
+              <ul *dropdownMenu class="dropdown-menu" aria-labelledby="unitsDropdown">
+                <li *ngFor="let unit of chart.indicator.available_units">
+                  <a (click)="onUnitSelected(unit)"
+                    placement="bottom">{{ unit }}</a>
+                </li>
+              </ul>
+            </div>`
+})
+export class UnitsDropdownComponent {
+
+    @Input() chart: Chart;
```

Comment:
The problem with the solution as implemented is that there's no way for the units dropdown component to tell its parent explicitly that a new unit value was selected -- `chart.unit` isn't 'watched'. My guess is that this is the reason for the extra `resetChart()` machinery below.

Happy to pair today to attempt a refactor if that would be helpful!